pkg/splunk/enterprise/util.go

```diff
@@ -913,16 +913,26 @@ func changePhaseInfo(desiredReplicas int32, appSrc string, appSrcDeployStatus ma
 				continue
 			}
 
+			// CSPL-1570: Set the AuxPhaseInfo to Install complete for all the existing pods.
+			// This is done so that we don't try to reinstall the app again on the existing pod.
+			if len(appDeployInfoList[idx].AuxPhaseInfo) == 0 &&
+				appDeployInfoList[idx].PhaseInfo.Phase == enterpriseApi.PhaseInstall &&
+				appDeployInfoList[idx].PhaseInfo.Status == enterpriseApi.AppPkgInstallComplete {
+				// Fill in the AuxPhaseInfo fields for current replicas
+				for i := 0; i < int(currentReplicas); i++ {
+					appDeployInfoList[idx].AuxPhaseInfo = append(appDeployInfoList[idx].AuxPhaseInfo, enterpriseApi.PhaseInfo{Phase: enterpriseApi.PhaseInstall, Status: enterpriseApi.AppPkgInstallComplete})
+				}
+			}
+
 			// set the phase to download
 			appDeployInfoList[idx].PhaseInfo.Phase = enterpriseApi.PhaseDownload
 
 			// set the status to download pending
 			appDeployInfoList[idx].PhaseInfo.Status = enterpriseApi.AppPkgDownloadPending
 
-			if len(appDeployInfoList[idx].AuxPhaseInfo) != 0 {
-				// update the aux phase info
-				updateAuxPhaseInfo(&appDeployInfoList[idx], desiredReplicas)
-			}
+			// update the aux phase info
+			updateAuxPhaseInfo(&appDeployInfoList[idx], desiredReplicas)
```

Comment:
Nit: you may want to rename to something like `createAuxPhaseInfoForNewPods() ` or something similar.